### ISSUE TITLE
"Predict" PV using future satellite images (as a 'best case' benchmark)

### ISSUE THREAD

**Issue by JackKelly** [openclimatefix/predict_pv_yield#53]
Just reshape satellite data to concat timesteps into the batch dimension.  And maybe then re-shape and use an RNN or something.

**Comment by JackKelly** [comment]
Better model:

Use smaller image:  just 64 x 64.  Use all satellite channels.

Feed each satellite image into a CNN.  That CNN also gets center marker, geo spatial coords, pixel coords, datetime features.

Also skip over CNN for a central square of satellite imagery (probably need pixel-level accuracy to really nail the timing - this is where self-attention on a per-pixel basis will probably be especially important)

Feed that into an RNN which also gets NWPs & PV system ID embedding.  Init the decoder RNN with an 'encoder' RNN which gets the recent history (perhaps as well as the output from the CNN, so it can compare recent 'predictions' with actual).

**Comment by JackKelly** [comment]
Training seems pretty unstable when simply stacking all the timesteps as different examples for the CNN.  Maybe pre-train the CNN on its own for a few epochs just one timestep per example, then swap.  Or only do entire sequences during validation?

**Comment by JackKelly** [comment]
yeah, probably want at least a 128 x 128 patch at low res, with a centre crop at high res, possibly just pass, say, an 8x8 pixel array straight to the fully connected layers.
